src/main/java/duke/Command.java

```diff
@@ -0,0 +1,298 @@
+     * @param ui The Object to render the UI
+     * @param tasks The Object to contain the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void searchAndDisplayTaskList(String searchDetails, UI ui, TaskList tasks) throws DukeException {
+        if (searchDetails != null && !searchDetails.equals("")) {
+            ui.printTaskList(tasks.searchTaskList(searchDetails));
+        } else {
+            throw new DukeException("OOPS! I'm sorry, but I don't know that command");
+        }
+    }
+    /**
+     * Method to check for the List Command and render the UI
+     *
+     * @param taskDetails The String describing the various attributes for the command
+     * @param ui The Object to render the UI
+     * @param tasks The Object to contain the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void displayTaskList(String taskDetails, UI ui, TaskList tasks) throws DukeException {
+        if (taskDetails != null && taskDetails.equals("")) {
+            ui.printTaskList(tasks);
+        } else {
+            throw new DukeException("OOPS! I'm sorry, but I don't know that command");
+        }
+    }
+
+    /**
+     * Method to check for the ToDO Command and render the UI
+     *
+     * @param taskDetails The String describing the various attributes for the command
+     * @param ui The Object to render the UI
+     * @param tasks The Object to contain the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void addTodo(String taskDetails, UI ui, TaskList tasks) throws DukeException {
+        if ((taskDetails == null) || (taskDetails.equals(""))) {
+            throw new DukeException("Incorrect Format of the ToDo Command!!, Correct Format --> todo <Description>");
+        } else {
+            ToDoTask todo = new ToDoTask(taskDetails);
+            tasks.addTask(todo);
+            ui.display("Got it. I've added this task:");
+            ui.display("  " + todo);
+            ui.display("Now you have " + tasks.getTaskListLength() + " tasks in the list.");
+        }
+    }
+
+    /**
+     * Method to check the type of the Command and distribute to the specific type of Command Handler.
+     *
+     * @param ui The Object to render the UI
+     * @param tasks The Object to contain the List of the Tasks
+     * @param storage The Object to save the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    public void execute(UI ui, TaskList tasks, Storage storage) throws DukeException {
+        switch (typeOfCommand) {
+        case BYE:
+            goodbye(taskDetails, ui);
+            storage.saveTaskList(tasks);
+            break;
+        case DEADLINE:
```

Comment:
Perhaps it would be good to name this ```ADD_DEADLINE```? I feel that ```DEADLINE``` is a bit confusing (same for ```EVENT``` and ```TODO```) since it doesn't sound like a command